boltz/db.go

```diff
@@ -293,26 +342,80 @@ func (self *DbImpl) AddRestoreListener(f func()) {
 	self.restoreListeners.Append(f)
 }
 
-func (self *DbImpl) SnapshotToMemory() (string, []byte, error) {
-	buf := &bytes.Buffer{}
-	id, err := self.SnapshotToWriter(buf)
-	return id, buf.Bytes(), err
+func (self *DbImpl) SnapshotToWriter(w io.Writer, requireTimelineReset bool) (string, error) {
+	var snapshotId string
+	err := self.Update(nil, func(ctx MutateContext) error {
+		var err error
+		snapshotId, err = self.SnapshotToWriterInTx(ctx.Tx(), w, requireTimelineReset)
+		return err
+	})
+	if err != nil {
+		return "", err
+	}
+	return snapshotId, nil
+}
+
+func (self *DbImpl) SerializeToWriter(w io.Writer) error {
+	return self.View(func(tx *bbolt.Tx) error {
+		_, err := tx.WriteTo(w)
+		return err
+	})
 }
 
-func (self *DbImpl) SnapshotToWriter(w io.Writer) (string, error) {
+func (self *DbImpl) SnapshotToWriterInTx(tx *bbolt.Tx, w io.Writer, requireTimelineReset bool) (string, error) {
 	snapshotId := uuid.NewString()
-	err := self.Update(nil, func(ctx MutateContext) error {
-		b := GetOrCreatePath(ctx.Tx(), Metadata)
-		b.SetString(SnapshotId, snapshotId, nil)
+	b := GetOrCreatePath(tx, Metadata)
+	b.SetString(SnapshotId, snapshotId, nil)
+
+	if requireTimelineReset {
+		b.SetBool(ResetTimeline, true, nil)
 		if b.HasError() {
-			return b.GetError()
+			return "", b.GetError()
 		}
-		_, err := ctx.Tx().WriteTo(w)
-		return err
-	})
+	}
+
+	pfxlog.Logger().WithField("writable", tx.Writable()).Infof("setting snapshot id: %s, reset timeline: %v", snapshotId, requireTimelineReset)
+
+	_, err := tx.WriteTo(w)
 	if err != nil {
 		return "", err
 	}
+
+	//if requireTimelineReset {
+	//	b.SetBool(ResetTimeline, false, nil)
+	//	if b.HasError() {
+	//		return "", b.GetError()
+	//	}
+	//}
```

Comment:
Dead code.

Reply:
Fixed. There were a full other uncommented, but unused methods that I also removed.